### DISCUSSION TITLE
Run my function on Nuxt 3 start

### DISCUSSION THREAD

**Discussion by topd5** [nuxt/nuxt#26263] [Questions]
Hi.
How to run any code on Nuxt app start?
Is any callback feature available, like in NodeJS app?
```
server.on('listening', () => { console.log('Server started'); });
server.listen(4321);
```
Code should run only once.
I can manually add some code in the beginning of `.output/server/index.mjs`, but this an ugly solution.

**Comment by xjccc** [comment]
see this.
[app-hooks](https://nuxt.com/docs/api/advanced/hooks#app-hooks-runtime)

**Comment by topd5** [comment]
Thanks for the answer.
But all this hooks don't help.
Nitoapp runtime hooks are: "close", "error", , "render:response", "request", "beforeResponse", "afterResponse".
There is no some 'listen' or 'start' hook here.

And 'listen' hook from "[Nuxt Hooks (build time)" is only for dev server.

**Reply by xjccc** [reply]
`app:created` cant use for you ?

**Reply by topd5** [reply]
No, this is vue app hook. It runs many times for every request.

**Comment by topd5** [comment]
Some simple workaround: use `defineNitroPlugin()` as some "server start'' hook:
```
export default defineNitroPlugin((nitroApp) => {
  console.log('Nuxt 3 server starting')
});
```

**Comment by huang-julien** [comment]
Hi :wave: maybe do you want to take a look at[ Nitro Task ](https://nitro.unjs.io/guide/tasks) and https://github.com/unjs/nitro/issues/1974

**Reply by DesertCookie** [reply]
While I find the idea of tasks intriguing, and the example shows a db migration task, it is not shown how to execute that task on startup like you would expect it to happen. I think the docs and references in [unjs/nitro#1974](https://github.com/unjs/nitro/issues/1974) fall short of explaining that.
